### PR TITLE
[APPC-1211] Add analytics to local payments

### PR DESCRIPTION
**Pull request by JDSM01**
**What does this PR do?**

   Added Local Payment analytics. 
Refactored Paypal analytics to avoid sending some of them when doing a local payment and a top up. 
Fixed CC analytics being sent when doing a purchase with paypal

**Where should the reviewer start?**

AdyenAuthorizationPresenter.java
LocalPaymentsPresenter.kt

**How should this be manually tested?**

Do a purchase with PayPal and Credit card (IaP and popup) and check if the events are being sent correctly on the console.
The local payments events will be later tested with Tiago. The only one we are able to check is the payment methods detail one and the payment event for the assync payments (alfamart)

**What are the relevant tickets?**

https://aptoide.atlassian.net/browse/APPC-1211

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/topup/payment/PaymentAuthPresenter.java [hunk 1]
```diff
@@ -249,14 +249,13 @@ private void handleAdyenUriResult() {
...
         .observeOn(viewScheduler)
         .filter(s -> !waitingResult)
         .doOnSuccess(redirectUrl -> {
           view.showLoading();
-          navigator.navigateToUriForResult(redirectUrl, billingService.getTransactionUid(), domain,
-              "", new BigDecimal(amount), transactionType);
+          navigator.navigateToUriForResult(redirectUrl);
           waitingResult = true;
         })
         .subscribe(__ -> {
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/topup/payment/PaymentAuthPresenter.java:261]: - onError method can be replaced with method reference.

- Please look at all the methods, the same applies for almost all of them

File: app/src/main/java/com/asfoundation/wallet/ui/iab/BillingWebViewFragment.java [hunk 2]
```diff
@@ -20,16 +20,12 @@
...
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.atomic.AtomicReference;
 import javax.inject.Inject;
 
-import static com.asfoundation.wallet.analytics.FacebookEventLogger.EVENT_REVENUE_CURRENCY;
-import static com.asfoundation.wallet.billing.analytics.BillingAnalytics.PAYMENT_METHOD_PAYPAL;
-
 public class BillingWebViewFragment extends DaggerFragment {
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/BillingWebViewFragment.java:29]:
```
  private static final String DOMAIN = "domain";
  private static final String SKUID = "skuid";
  private static final String AMOUNT = "amount";
  private static final String TYPE = "type";
```

They are not used and can be removed

File: app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentAnalytics.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package com.asfoundation.wallet.ui.iab
+
+import com.asfoundation.wallet.analytics.FacebookEventLogger.EVENT_REVENUE_CURRENCY
+import com.asfoundation.wallet.billing.analytics.BillingAnalytics
+import io.reactivex.Scheduler
+import io.reactivex.disposables.CompositeDisposable
+import java.math.BigDecimal
+
+class LocalPaymentAnalytics(private val analytics: BillingAnalytics,
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentAnalytics.kt:9]: Code style on function arguments can be improved

File: app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentInteractor.kt [hunk 4]
```diff
@@ -55,10 +55,10 @@ class LocalPaymentInteractor(private val deepLinkRepository: InAppDeepLinkReposi
     return (status == PENDING_USER_PAYMENT && type == "TOPUP") || (status == COMPLETED && (type == "INAPP" || type == "INAPP_UNMANAGED")) || status == FAILED || status == CANCELED || status == INVALID_TRANSACTION
   }
 
-  fun getCompletePurchaseBundle(isInApp: Boolean, merchantName: String, sku: String?,
+  fun getCompletePurchaseBundle(type: String, merchantName: String, sku: String?,
                                 scheduler: Scheduler,
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentInteractor.kt]: the readability in this method could be improved if we organize the arguments in a different way

File: app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentInteractor.kt [hunk 5]
```diff
@@ -55,10 +55,10 @@ class LocalPaymentInteractor(private val deepLinkRepository: InAppDeepLinkReposi
...
                                 orderReference: String?, hash: String?): Single<Bundle> {
-    return if (isInApp && sku != null) {
+    return if (isInApp(type) && sku != null) {
       billing.getSkuPurchase(merchantName, sku, scheduler)
           .map {
             billingMessagesMapper.mapPurchase(it,
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentInteractor.kt]: The code inside the map function could be simplified to a single line

File: app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentPresenter.kt [hunk 6]
```diff
@@ -6,6 +6,7 @@ import com.appcoins.wallet.bdsbilling.repository.entity.Transaction.Status
 import io.reactivex.Completable
 import io.reactivex.Scheduler
 import io.reactivex.disposables.CompositeDisposable
+import java.math.BigDecimal
 import java.util.concurrent.TimeUnit
 
 private val WAITING_RESULT = "WAITING_RESULT"
```
Comment (carvalho2707) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentPresenter.kt]: I think we can move this 2 constants to a companion object right?

**Review by carvalho2707 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentInteractor.kt (hunk 4 above)
Comment (JDSM01) [reply app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentInteractor.kt]: I’m only seeing changing the scheduler to last? is that it?

File: app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentInteractor.kt (hunk 5 above)
Comment (JDSM01) [reply app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentInteractor.kt]: done

**Review by carvalho2707 — APPROVED** [review]

**Pull request merged**